### PR TITLE
stackrox: acs step operator install

### PR DESCRIPTION
**Pull request by davdhacs**
ACS install step for OpenShift clusters
1. with rhacs-operator:stable and no external config or scripts.
2. central and secured-cluster on a single cluster provided with KUBECONFIG

Tested in pr https://github.com/openshift/release/pull/51002 passing [run](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51002/rehearse-51002-pull-ci-stackrox-stackrox-master-ocp-latest-cvp-install/1787968397275828224): [build-log.txt](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/51002/rehearse-51002-pull-ci-stackrox-stackrox-master-ocp-latest-cvp-install/1787968397275828224/artifacts/install/stackrox-install/build-log.txt)

### CODE REVIEW

File: ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        memory: 1Gi
+        cpu: 500m
+      limits:
+        memory: 4Gi
+        cpu: 1
+    exposure:
+      loadBalancer:
+        enabled: false
+      route:
+        enabled: false
```
Comment (gavin-stackrox) [inline ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh:81]: Should it be `true`? How will tests access central?
```suggestion
        enabled: true
```

File: ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+  url=https://github.com/jqlang/jq/releases/download/jq-1.7.1/jq-linux-amd64
+  echo "Downloading jq binary from ${url}"
+  curl -Ls -o ./jq "${url}"
+  chmod u+x ./jq
+  export PATH=${PATH}:${PWD}
+}
+jq --version || get_jq
+
+ROX_PASSWORD=$(oc -n stackrox get secret admin-pass -o json 2>/dev/null | jq -er '.data["password"] | @base64d') \
+  || ROX_PASSWORD="$(LC_ALL=C tr -dc _A-Z-a-z-0-9 < /dev/urandom | head -c12 || true)"
```
Comment (gavin-stackrox) [inline ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh:56]: Should this be saved to `${SHARED_DIR}/rox_password` or something for future steps to use to access central?

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]
This is great. I'd like to see a nightly job that exercises this to keep it viable for users. It could have a post install step that uses the route+rox_password to curl centrals metadata API.

File: ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    resources:
+      requests:
+        memory: 1Gi
+        cpu: 500m
+      limits:
+        memory: 4Gi
+        cpu: 1
+    exposure:
+      loadBalancer:
+        enabled: false
```
Comment (jtaleric) [inline ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh:79]: I would recommend template this out and reference the `true/false` aspects from the ref file.

File: ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh (hunk 1 above)
Comment (jtaleric) [reply ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh:81]: same comment as above, I would template this out.

File: ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+          cpu: 500m
+        limits:
+          memory: 4Gi
+          cpu: 1
+    telemetry:
+      enabled: false
+  scanner:
+    analyzer:
+      scaling:
+        autoScaling: Disabled
```
Comment (jtaleric) [inline ci-operator/step-registry/stackrox/install/stackrox-install-commands.sh:95]: same here - much of the bit flipping can happen at the job configuration.

**Comment by jtaleric** [comment]
> @jtaleric This is for adding the step-registry operator install of ACS. I tested it under the generic-claim workflow. Can you add it on a perf@scale config to test it there?

Did you want me to add this in this PR or on a follow on?

**Comment by gavin-stackrox** [comment]
> @gavin-stackrox are you okay with this as-is and I'll make the configuration change-able in a follow-up pr?

/lgtm

**Pull request merged**